crates/wasmparser/src/readers/core/types.rs

```diff
@@ -798,6 +859,19 @@ pub struct StructType {
     pub fields: Box<[FieldType]>,
 }
 
+impl StructType {
+    #[cfg(all(feature = "validate", feature = "component-model"))]
+    pub(crate) fn desc(&self) -> String {
+        let mut s = String::from("(struct");
+        for f in &self.fields {
+            s.push(' ');
+            s.push_str(&f.desc());
+        }
+        s.push_str(")");
+        s
+    }
+}
```

Comment:
Could this be done as `fmt::Display` or similar to avoid the intermediate string allocations?

Reply:
I was going to move all the `desc` (including ones already defined before this PR) to `Display` impls in a follow up. Would you prefer that happen in this PR?